src/vsr/checkpoint_trailer.zig

```diff
@@ -70,18 +74,18 @@ pub fn FreeSetEncodedType(comptime Storage: type) type {
         };
 
         // Reference to the grid is late-initialized in the open, because the free set is part of
-        // the superblock, which doesn't have access to grid. It is set to null by reset, to verify
-        // that the free set is not used before it is opened during sync.
+        // the grid, which doesn't have access to a stable grid pointer. It is set to null by
+        // `reset`, to verify that the free set is not used before it is opened during sync.
```

Comment:
some residual free sets in the comment here

Reply:
I think in this particular case the comment is correct -- the reason for lazy-initializing the grid pointer is specifically due to the free-set/grid relationship, not client sessions.